SU-SWS/react_paragraphs
# D8CORE-2150: Adjust link widget to mirror the php link widget

# READY FOR REVIEW

# Summary
- Mirrored the functions to their PHP counterparts with a little extra in there.

# Need Review By (Date)
- 10/2

# Urgency
- medium

# Steps to Test
1. Checkout this branch & clear browser caches
1. Add a paragraph that has a link field
1. paste a url
1. validate the url you pasted is unchanged, unless it a local url in which it converts it to relative.
1. try some other urls and some autocomplete suggestions to verify it all works and saves correctly.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
